services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeMapper.kt

```diff
@@ -0,0 +1,84 @@
+package fi.hel.haitaton.hanke
+
+import fi.hel.haitaton.hanke.ContactType.MUU
+import fi.hel.haitaton.hanke.ContactType.OMISTAJA
+import fi.hel.haitaton.hanke.ContactType.RAKENNUTTAJA
+import fi.hel.haitaton.hanke.ContactType.TOTEUTTAJA
+import fi.hel.haitaton.hanke.domain.Hanke
+import fi.hel.haitaton.hanke.domain.HankeYhteystieto
+import fi.hel.haitaton.hanke.domain.Hankealue
+import fi.hel.haitaton.hanke.geometria.Geometriat
+import fi.hel.haitaton.hanke.tormaystarkastelu.TormaystarkasteluTulos
+
+object HankeMapper {
+
+    /**
+     * Needs to be called inside a transaction for lazy fields to be accessible.
+     *
+     * @param entity Source hanke entity.
+     * @param geometriaData Since hanke only has an id reference on the actual geometries.
+     */
+    fun domainFrom(entity: HankeEntity, geometriaData: Map<Int, Geometriat?>): Hanke =
+        with(entity) {
+            Hanke(
+                    id = id,
+                    hankeTunnus = hankeTunnus,
+                    onYKTHanke = onYKTHanke,
+                    nimi = nimi,
+                    kuvaus = kuvaus,
+                    vaihe = vaihe,
+                    suunnitteluVaihe = suunnitteluVaihe,
+                    version = version,
+                    createdBy = createdByUserId,
+                    createdAt = createdAt?.zonedDateTime(),
+                    modifiedBy = modifiedByUserId,
+                    modifiedAt = modifiedAt?.zonedDateTime(),
+                    status = status,
+                    generated = generated
+                )
+                .apply {
+                    val contacts = contacts(entity)
+                    omistajat = contacts[OMISTAJA].orEmpty().toMutableList()
+                    rakennuttajat = contacts[RAKENNUTTAJA].orEmpty().toMutableList()
+                    toteuttajat = contacts[TOTEUTTAJA].orEmpty().toMutableList()
+                    muut = contacts[MUU].orEmpty().toMutableList()
+                    tyomaaKatuosoite = entity.tyomaaKatuosoite
+                    tyomaaTyyppi = entity.tyomaaTyyppi
+                    alueet = alueList(entity.hankeTunnus, entity.listOfHankeAlueet, geometriaData)
+                    tormaystarkasteluTulos =
+                        tormaystarkasteluTulokset.firstOrNull()?.let {
+                            TormaystarkasteluTulos(it.perus, it.pyoraily, it.joukkoliikenne)
+                        }
+                }
+        }
+
+    private fun contacts(entity: HankeEntity): Map<ContactType, MutableList<HankeYhteystieto>> =
```

Comment:
Making the lists mutable isn't necessary here, since `.toMutableList()` needs to be called anyway after `.orEmpty()`.

Reply:
Fixed.